test/e2e/kubectl/kubectl.go

```diff
@@ -1507,11 +1507,23 @@ metadata:
 			Description: The command 'kubectl version' MUST return the major, minor versions,  GitCommit, etc of the Client and the Server that the kubectl is configured to connect to.
 		*/
 		framework.ConformanceIt("should check is all data is printed ", func() {
-			version := framework.RunKubectlOrDie(ns, "version")
+			// windows newline to \n
+			versionString := strings.Replace(framework.RunKubectlOrDie(ns, "version"), "\r\n", "\n", 1)
+			version := strings.FieldsFunc(versionString, splitVersionString)
 			requiredItems := []string{"Client Version:", "Server Version:", "Major:", "Minor:", "GitCommit:"}
```

Comment:
I was actually expecting something simpler, along the lines of:
```go
requiredItems := []string{"Client Version: ", "Server Version: "}
for _, item := range requiredItems {
	if matched, _ := regexp.MatchString(item+`version.Info\{Major:"\d", Minor:"[\d\+]+", GitVersion:"v\d\.\d+.[\w\-\.\+]+", GitCommit:"[0-9a-f]+"`, output); !matched {
		fmt.Printf("didn't match %s\n", item)
	}
}
```

You could tweak the regexp and/or items, but that should be sufficient.

Reply:
Will update

Reply:
Previously, I wasn't sure if the order of items Major, Minor, GitVersion,... will be in same order always, so i split the string to check. I have it updated now.